### ISSUE TITLE
DETACH PARTITION vs merges = duplicates

### ISSUE THREAD

**Issue by filimonov** [ClickHouse/ClickHouse#7415]
test case:
```
create table test2 Engine=MergeTree ORDER BY number AS SELECT number, toString(rand()) x from numbers(100000000);
alter table test2 detach partition tuple();
detach table test2; attach table test2; -- or just restart clickhouse
alter table test2 attach partition tuple();
select count() from test2;
``` 

Expected 100000000, got 200000000

**Comment by excitoon** [comment]
I got:
```
select count() from test2;

SELECT count()
FROM test2

┌───count()─┐
│ 199614720 │
└───────────┘
```

**Comment by filimonov** [comment]
Fixed in #8602
